hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -950,18 +959,57 @@ class IRNode:
     value: IRType = attr.ib()
     kwargs: IRKWargsType = attr.ib()
     was_forced: bool = attr.ib()
+    index: Optional[int] = attr.ib(default=None)
 
     def copy(self, *, with_value: IRType) -> "IRNode":
         # we may want to allow this combination in the future, but for now it's
         # a footgun.
         assert not self.was_forced, "modifying a forced node doesn't make sense"
+        # explicitly not copying index. node indices are only assigned via
+        # ExampleRecord. This prevents footguns with relying on stale indices
+        # after copying.
         return IRNode(
             ir_type=self.ir_type,
             value=with_value,
             kwargs=self.kwargs,
             was_forced=self.was_forced,
         )
 
+    @property
+    def trivial(self):
+        """
+        A node is trivial if it cannot be simplified any further. This does not
+        mean that modifying a trivial node can't produce simpler test cases when
+        viewing the tree as a whole. Just that when viewing this node in
+        isolation, this is the simplest the node can get.
+        """
+        if self.was_forced:
+            return True
+
+        if self.ir_type == "integer":
+            return self.value == self.kwargs["min_value"]
+        if self.ir_type == "float":
+            return float_to_int(self.value) == float_to_int(self.kwargs["min_value"])
+        if self.ir_type == "boolean":
+            return self.value is False
+        if self.ir_type == "string":
+            # we can do better here with "length is equal to min_size and value is
+            # filled with simplest-in-shrink-order characters". This requires
+            # computing what the simplest character in a given IntervalSet is,
+            # and since we redefine the shrink order for characters, that requires
+            # computing the order for all characters in an IntervalSet. Which
+            # may be expensive for large intervals. (would @cached_property be
+            # good enough here?)
+            #
+            # This naive check is hopefully good enough for now? I'm trying to
+            # avoid doing more work than we save when computing `trivial`.
+            return self.value == ""
```

Comment:
```suggestion
            minimal_char = self.kwargs["intervals"].char_in_shrink_order(0)
            return self.value == (minimal_char  * self.kwargs["min_size"])
```
I think this works and should be efficient - it's just some integer arithmetic and comparisons.

Reply:
Nice, thanks! Missed `char_in_shrink_order(0)`.